### DISCUSSION TITLE
Wondering about the performance

### DISCUSSION THREAD

**Discussion by danielius-tvarijonas** [uxbox/uxbox#1371] [General]
Hi, I have a question about the performance  of the latest version of the Penpot.
I have a 3 MB document that consist of 4 artboards (1920 x 3000px) with < 150 layers. When designing I'm experiencing a very noticeable sluggishness/delay when moving around the canvas, when selecting and interacting (moving, changing parameters, applying layer effects, etc) with layers. I'm running it on the Docker and my machine is relatively powerful (2019 iMac: 6-core CPU, 40GB of RAM; browser: Brave), I also tried the https://design.penpot.app/ but the experience is the same. So, I'm wondering whether it is a normal behaviour since it still is a beta product or whether there's something I could do to fix it?

**Comment by hirunatan** [comment]
Hi, @john777555. The performance of editing a document with many layers is a quite complex issue. We are still improving it, in the next release you may expect significative enhancements of this. 

Not only the quantity of layers impacts on the speed. In particular some effects like gradients or shadows have an important drawing cost, for example.

**Reply by danielius-tvarijonas** [reply]
Thanks, can't wait for the next release.

**Reply by m-collins** [reply]
Performance is still a major issue unfortunately. Really wanting to see Penpot do well but current performance is a deal breaker for me right now. Every interaction is slightly delayed on a relatively simple project.

**Reply by diacritica** [reply]
Hi! We're getting very near to Penpot 2.0 now (just a few weeks away) and performance improvements are a major part of it, perhaps you can try it when it's published. Early reports from beta testers are very explicit about how much smoother the whole Penpot platform is.

**Reply by budislav** [reply]
Performance is still very bad on 2.0. This is only reason why we can't use Penpot for daily work, even for personal project. When you move around zoom and even draw around many layers are impossible. I am using MacBook pro m1 with 16gb on Chrome, performance is so bad. Maybe Penpot need refactoring or something but this is the most important thing I think because we can not use it in this condition.

For example try to open Gnome components and navigate to window section, than zoom in and try to pan canvas with middle button. And this section is not the most complex thing you will make, this is basic thing with small amount of layers. 

So at this moment because of performance, it is not usable, maybe for small application with 5-10 screens without components, but not for design systems with large amount of components, and that is the way we are designing today, everything is in components, so naturally we make huge amount of layers. Please optimize Penpot.

**Comment by diacritica** [comment]
Hey @john777555 see this too #1482

**Reply by danielius-tvarijonas** [reply]
hey @diacritica 
Thanks, I'd definitely want to see how you're dealing with it as I still find it difficult to use when designing for desktop resolutions 😭, works quite well for mobile projects though where the artboard sizes are of a significantly smaller dimensions.

**Reply by diacritica** [reply]
There is an upcoming release that deals with both performance and zoom options, I wonder if you could test it and give some feedback on your particular use case. Remember, you can share your designs (if you want) to allow us to troubleshoot the "real thing"

**Reply by danielius-tvarijonas** [reply]
Sure, I'd love to do that! Is there an ETA for it?

**Reply by diacritica** [reply]
The team is doing final QA this week so it's close, but we don't have a date yet. Watch out for any live streaming related to that, I think there's one early next week.

**Comment by cyberalt1** [comment]
This is the single biggest issue of Penpot so far, if this is fixed Figma would be replaced overnight.

**Reply by girafic** [reply]
@cyberalt1 Penpot is already working on the new render engine --> https://community.penpot.app/t/its-time-for-penpot-to-almost-move-away-from-the-dom/6437
